api/v1alpha2/backstage_types.go

```diff
@@ -0,0 +1,302 @@
+//
+// Copyright (c) 2023 Red Hat, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha2
+
+import (
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+)
+
+type BackstageConditionReason string
+
+type BackstageConditionType string
+
+const (
+	BackstageConditionTypeDeployed BackstageConditionType = "Deployed"
+
+	BackstageConditionReasonDeployed   BackstageConditionReason = "Deployed"
+	BackstageConditionReasonFailed     BackstageConditionReason = "DeployFailed"
+	BackstageConditionReasonInProgress BackstageConditionReason = "DeployInProgress"
+)
+
+// BackstageSpec defines the desired state of Backstage
+type BackstageSpec struct {
+	// Configuration for Backstage. Optional.
+	Application *Application `json:"application,omitempty"`
+
+	// Raw Runtime RuntimeObjects configuration. For Advanced scenarios.
+	RawRuntimeConfig *RuntimeConfig `json:"rawRuntimeConfig,omitempty"`
+
+	// Configuration for database access. Optional.
+	Database *Database `json:"database,omitempty"`
+
+	// Valid fragment of Deployment to be merged with default/raw configuration.
+	// Set the Deployment's metadata and|or spec fields you want to override or add.
+	// Optional.
+	// +kubebuilder:pruning:PreserveUnknownFields
+	Deployment *apiextensionsv1.JSON `json:"deployment,omitempty"`
```

Comment:
Would it make sense to call it `deploymentPatch` or `deploymentOverride` or something like this to make it clear that this is not the full deployment definition?

Reply:
Makes sense.
Then, I would propose to make it as
````
deployment:
     patch:
````

this way we can also consider similar feature for local database (as I understand we'd rather want to have it editable too)
````
database:
     enableLocalDb:
     authSecretName:
     patch:
````
 
 WDYT?

Reply:
did it this way